romancal/pipeline/exposure_pipeline.py

```diff
@@ -71,66 +75,100 @@ def process(self, input):
+        file_type = filetype.check(input)
+        asn = None
+        if file_type == "asdf":
+            try:
+                input = rdm.open(input)
+            except TypeError:
+                log.debug("Error opening file:")
+        if file_type == "asn":
+            try:
+                asn = LoadAsLevel2Asn.load(input, basename=self.output_file)
+            except AssociationNotValidError:
+                log.debug("Error opening file:")
+
+        # Build a list of observations to process
+        expos_file = []
+        if file_type == "asdf":
+            expos_file = [input]
+        elif file_type == "asn":
+            for product in asn["products"]:
+                for member in product["members"]:
+                    expos_file.append(member["expname"])
+
+        results = []
+        for in_file in expos_file:
+            if isinstance(in_file, str):
+                input_filename = basename(in_file)
+                log.info(f"Input file name: {input_filename}")
+            else:
+                input_filename = None
+
+            # Open the file
+            input = rdm.open(in_file)
+            log.info(f"Processing a WFI exposure {in_file}")
+
+            self.dq_init.suffix = "dq_init"
+            result = self.dq_init(input)
+            if input_filename:
+                result.meta.filename = input_filename
+            result = self.saturation(result)
+
+            # Test for fully saturated data
             if is_fully_saturated(result):
-                # Set all subsequent steps to skipped
-                for step_str in [
-                    "assign_wcs",
-                    "flat_field",
-                    "photom",
-                    "source_detection",
-                ]:
-                    result.meta.cal_step[step_str] = "SKIPPED"
-
-                # Set suffix for proper output naming
-                self.suffix = "cal"
+                log.info("All pixels are saturated. Returning a zeroed-out image.")
+
+                # Return zeroed-out image file (stopping pipeline)
+                return self.create_fully_saturated_zeroed_image(result)
+
+            result = self.linearity(result)
+            result = self.dark_current(result)
+            result = self.jump(result)
+            result = self.rampfit(result)
+
+            # Test for fully saturated data
+            if "groupdq" in result.keys():
+                if is_fully_saturated(result):
+                    # Set all subsequent steps to skipped
+                    for step_str in [
+                        "assign_wcs",
+                        "flat_field",
+                        "photom",
+                        "source_detection",
+                    ]:
+                        result.meta.cal_step[step_str] = "SKIPPED"
+
+                    # Set suffix for proper output naming
+                    self.suffix = "cal"
+                    results.append(result)
```

Comment:
It feels to me like this block never executes, since it's below the return self.create_fully_saturated... bit above.  I suspect you want to get rid of this block and incorporate the 'SKIPPED' bit into the earlier block?

Reply:
See above